torchvision/csrc/io/image/cpu/encode_png.cpp

```diff
@@ -155,7 +150,7 @@ torch::Tensor encode_png(const torch::Tensor& data, int64_t compression_level) {
   auto ptr = input.data_ptr<uint8_t>();
 
   // Encode PNG file
-  for (size_t y = 0; y < height; ++y) {
+  for (int y = 0; y < height; ++y) {
```

Comment:
instead of making `y` `int`, wouldn't it be better to make `height` (and `width`) to be `int64_t` (as it is set by default in PyTorch) ?

Reply:
I avoided changing `height` and `width` because they are passed around to other methods (some of which we don't have control such as `png_set_IHDR`) and there they would require downcasting. Hence I opted for the implementation that would require less changes on the code-base. 

Perhaps another future Nit-fixing PR can replace all size types in a consistent manner across the codebase. In practice all should be equivalent because the image size will fit int 32.